src/main/java/org/broadinstitute/hellbender/engine/GATKTool.java

```diff
@@ -585,7 +589,7 @@ private void validateSequenceDictionaries() {
 
         final SAMSequenceDictionary refDict = hasReference() ? reference.getSequenceDictionary() : null;
         final SAMSequenceDictionary readDict = hasReads() ? reads.getSequenceDictionary() : null;
-        final List<SAMSequenceDictionary> featureDicts = hasFeatures() ? features.getAllSequenceDictionaries() : Collections.emptyList();
+        final List<SAMSequenceDictionary> featureDicts = hasFeatures() ? features.getAllSequenceDictionaries(errorOnOutOfDateIndex) : Collections.emptyList();
```

Comment:
Come to think of it, I actually don't think this check should be embedded in the sequence dictionary getters like this. It should happen in the `FeatureDataSource` constructor(s) instead.